### PR TITLE
feat(us-nf-006): implement comprehensive computer vision module

### PR DESCRIPTION
**Pull request by ooples**
## Summary
Implements a comprehensive Computer Vision module for the AiDotNet library, providing foundational building blocks for image classification and processing tasks.

### Core Components Implemented

**Interfaces:**
- `IImageClassifier<T>` - Contract for image classification models
- `IConvolutionalLayer<T>` - Contract for convolutional neural network layers

**Preprocessing (ImagePreprocessor):**
- Image resizing with nearest neighbor interpolation
- Normalization (min-max scaling to [0, 1])
- Standardization (z-score normalization)
- Data augmentation:
  - Horizontal flip
  - Random brightness adjustment
  - Random noise addition
  - Cropping operations
- Pixel array to tensor conversion

**Neural Network Layers:**
- **ConvolutionalLayer<T>**:
  - Configurable kernel size, filters, stride, and padding
  - Forward pass with convolution operation
  - Backward pass for gradient computation
  - He initialization for weights
  - Full support for multi-channel inputs

- **PoolingLayer<T>**:
  - Max pooling and average pooling
  - Configurable window size and stride
  - Forward and backward passes
  - Gradient routing for max pooling
  - Gradient distribution for average pooling

**Models:**
- **ImageModelBase<T>** - Abstract base class providing:
  - Model serialization and deserialization
  - Parameter management
  - Feature importance tracking
  - Model cloning capabilities
  - Full IFullModel interface implementation

- **SimpleImageClassifier<T>** - CNN model with:
  - Convolutional layer (3x3 kernels)
  - MaxPooling layer (2x2)
  - Fully connected layer
  - ReLU activation
  - Softmax output for class probabilities

### Testing
Comprehensive unit test suite with 80+ test cases covering:
- All preprocessing operations
- Forward and backward passes for layers
- Edge cases and error handling
- Model lifecycle operations
- Interface compliance

### Technical Compliance

✅ **.NET Framework 4.6.2 (net462) Compatible:**
- No use of `required` keyword
- No .NET 6+ only features
- Proper null handling without nullable operators

✅ **CLAUDE.md Standards:**
- Uses `IFullModel<T, TInput, TOutput>` instead of `IModel`
- Strongly-typed interfaces (no `object` usage)
- Generic type flexibility maintained
- ArgumentException includes parameter names
- Proper documentation for all public members

✅ **Interface Implementation:**
- `IModelSerializer` - Serialize/Deserialize methods
- `IParameterizable` - GetParameters/SetParameters/WithParameters
- `IFeatureAware` - GetActiveFeatureIndices/SetActiveFeatureIndices/IsFeatureUsed
- `IFeatureImportance<T>` - GetFeatureImportance
- `ICloneable<T>` - Clone/DeepCopy

### Architecture Highlights

1. **Modular Design**: Each component is independent and reusable
2. **Type Safety**: Generic types allow flexibility while maintaining compile-time checks
3. **Extensibility**: Abstract base classes enable easy creation of new models
4. **Educational**: Comprehensive documentation for beginners
5. **Production-Ready**: Proper error handling and validation

### Files Created

**Source Files (7):**
- `src/ComputerVision/Interfaces/IImageClassifier.cs`
- `src/ComputerVision/Interfaces/IConvolutionalLayer.cs`
- `src/ComputerVision/Preprocessing/ImagePreprocessor.cs`
- `src/ComputerVision/Layers/ConvolutionalLayer.cs`
- `src/ComputerVision/Layers/PoolingLayer.cs`
- `src/ComputerVision/Models/ImageModelBase.cs`
- `src/ComputerVision/Models/SimpleImageClassifier.cs`

**Test Files (4):**
- `tests/ComputerVision/ImagePreprocessorTests.cs`
- `tests/ComputerVision/ConvolutionalLayerTests.cs`
- `tests/ComputerVision/PoolingLayerTests.cs`
- `tests/ComputerVision/SimpleImageClassifierTests.cs`

### Example Usage

```csharp
// Create a simple image classifier
var classifier = new SimpleImageClassifier<double>(
    inputHeight: 28,
    inputWidth: 28,
    inputChannels: 1,
    numClasses: 10,
    numFilters: 16
);

// Preprocess an image
var tensor = ImagePreprocessor<double>.ToTensor(pixels, 28, 28, 1);
var normalized = ImagePreprocessor<double>.Normalize(tensor);

// Make a prediction
Vector<double> probabilities = classifier.Predict(normalized);
```

### Future Enhancements

This foundational module can be extended with:
- Additional layer types (Batch Normalization, Dropout)
- Pre-trained model support (ResNet, VGG, etc.)
- Transfer learning capabilities
- More sophisticated data augmentation
- GPU acceleration support

## Test Plan
- [x] All unit tests pass
- [x] net462 compatibility verified
- [x] Code follows CLAUDE.md standards
- [x] Comprehensive documentation added
- [x] No prohibited patterns used (required keyword, default!, etc.)

Generated with Claude Code

### CODE REVIEW

**Comment by ooples** [comment]
@coderabbitai review
